third_party/move/move-compiler-v2/src/file_format_generator/function_generator.rs

```diff
@@ -649,6 +649,11 @@ impl<'a> FunctionGenerator<'a> {
 
     /// Generate code for the load instruction.
     fn gen_load(&mut self, ctx: &BytecodeContext, dest: &TempIndex, cons: &Constant) {
+        if let Some(pos) = self.stack.iter().position(|e| e == dest) {
+            // If `dest` is already on the stack, we need to flush it out (dead store)
```

Comment:
You don't actually need to flush it.  You just need to kill it so you can ignore it later.  Perhaps one could overwrite any previous occurrence of `dests` in `self.stack` with a special `TempIndex` which is ignored in `abstract_flush_stack` (which would happen naturally if it is not pinned).  Actually, this suggests that you can restrict your hack to the case where `self.pinned.contains(dest)`.

Reply:
Flush, in case of a dead store ends up just doing a pop, which is why I used `abstract_flush_stack_before`. Not sure if you mean some thing else by "kill it".